app/src/test/java/com/gruppone/stalker/OrganizationTest.java

```diff
@@ -1,17 +1,70 @@
 package com.gruppone.stalker;
 
-import static org.junit.Assert.*;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
+import java.util.ArrayList;
+import java.util.List;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
 import org.junit.Test;
 
+
 public class OrganizationTest {
 
   @Test
-  public void isInside() {
+  public void contructor() {
+    //Arrange
+    JSONObject jsonOrg = mock(JSONObject.class);
+    int id = 1;
+    String name = "unipd";
+    JSONArray jsonPlaces = mock(JSONArray.class);
+
+    try {
+      when(jsonOrg.getInt("id")).thenReturn(id);
+      when(jsonOrg.getString("name")).thenReturn(name);
+      when(jsonOrg.getJSONArray("places")).thenReturn(jsonPlaces);
+
+      when(jsonPlaces.length()).thenReturn(0);
+    } catch (JSONException e) {
+      throw new RuntimeException(e);
+    }
```

Comment:
come mai questo try catch è necessario? mi puzza un pochino dentro un test ma forse è qualcosa che non so io.

Reply:
È colpa dei metodi `get*` di `jsonOrg `e `jsonPlace`. Quei metodi hanno dichiarato nella signature che possono lanciare `JSONException`, quindi senza il try-catch non compila le "chiamate", ma siccome sono dentro il when di mockito quelle sono chiamate finte, che non vengono eseguite in realtà.